LanguageFeatures/Patterns/record_A04_t01.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// recordPattern         ::= '(' patternFields? ')'
+/// patternFields         ::= patternField ( ',' patternField )* ','?
+/// patternField          ::= ( identifier? ':' )? pattern
+/// ...
+/// It is a compile-time error if a name cannot be inferred for a named field
+/// pattern with the field name omitted
```

Comment:
This error is concerned with the case where the name of a named field whose existence is implied by a `<patternField>` that includes `:` cannot be taken from the part after the `:`.

For instance, `: 31` uses a `<constantPattern>`, and we can't know what the name of the field should be.

I think this means that there is no way we can have that situation in an irrefutable pattern: They must have a `<variablePattern>` after `:` (possibly with a unary pattern around them), so we can _always_ see a name.

Reply:
I added recorrd_A04_t03.dart where we check that it is a compile-time error if after `:` we have nor variable, nor null check/assert, nor cast or parethesized pattern. But, please, clarify what to do with this test (and the next one)? I updated the description to reflect what test really does, but didn't find in a spec that we really should expect compile-time error in this case. So, may bw these tests should be simply deleted